packages/node_modules/@ciscospark/plugin-memberships/src/memberships.js

```diff
@@ -24,6 +36,146 @@ import {SparkPlugin, Page} from '@ciscospark/spark-core';
+    return this.spark.internal.mercury.connect()
+      .then(() => this.listenTo(this.spark.internal.mercury,
+        WEBEX_EVENT.TEAMS_ACTIVITY,
+        (event) => this.onWebexApiEvent(event)));
+  },
+
+  /**
+   * Trigger a membership related events.
+   * @param {Object} event
+   * @returns {undefined}
+   */
+  onWebexApiEvent(event) {
+    const {activity} = event.data;
+
+    /* eslint-disable no-case-declarations */
+    switch (activity.verb) {
+      case API_ACTIVITY_VERB.ADD:
+        const membershipCreatedEventData =
+          this.getMembershipEvent(activity, SDK_EVENT.CREATED);
+
+        debug(`membership "created" payload: \
+          ${JSON.stringify(membershipCreatedEventData)}`);
+        this.trigger(SDK_EVENT.CREATED, membershipCreatedEventData);
+        break;
+
+      case API_ACTIVITY_VERB.LEAVE:
+        const membershipDeletedEventData =
+          this.getMembershipEvent(activity, SDK_EVENT.DELETED);
+
+        debug(`membership "deleted" payload: \
+          ${JSON.stringify(membershipDeletedEventData)}`);
+        this.trigger(SDK_EVENT.DELETED, membershipDeletedEventData);
+        break;
+
+      case API_ACTIVITY_VERB.ADD_MODERATOR:
+      case API_ACTIVITY_VERB.REMOVE_MODERATOR:
+        const membershipUpdatedEventData =
+          this.getMembershipEvent(activity, SDK_EVENT.UPDATED);
+
+        debug(`membership "updated" payload: \
+          ${JSON.stringify(membershipUpdatedEventData)}`);
+        this.trigger(SDK_EVENT.UPDATED, membershipUpdatedEventData);
+        break;
+
+      case API_ACTIVITY_VERB.ACKNOWLEDGE:
+        const membershipSeenEventData =
+          this.getMembershipEvent(activity, SDK_EVENT.SEEN);
+
+        debug(`membership "updated" payload: \
+          ${JSON.stringify(membershipSeenEventData)}`);
+        this.trigger(SDK_EVENT.SEEN, membershipSeenEventData);
+        break;
+
+      default:
+        break;
+    }
+  },
+
+  /**
+   * Constructs the data object for an event on the memberships resource,
+   * adhering to Hydra's Webehook data structure memberships.
+   * @param {Object} activity from mercury
+   * @param {Object} event type of "webhook" event
+   * @returns {Object} constructed event
+   */
+  getMembershipEvent(activity, event) {
+    const sdkEvent = JSON.parse(JSON.stringify(this.eventEnvelope));
+    let member = '';
+
+    sdkEvent.event = event;
+    sdkEvent.data.created = activity.published;
+    sdkEvent.actorId = constructHydraId(hydraTypes.PEOPLE, activity.actor.id);
+    sdkEvent.data.roomId =
+      constructHydraId(hydraTypes.ROOM, activity.target.id);
+    sdkEvent.data.roomType =
+      activity.target.tags.includes('ONE_ON_ONE') ? 'direct' : 'group';
+    if ((activity.object.roomProperties) && (activity.object.roomProperties.isModerator)) {
+      sdkEvent.data.isModerator = true;
+    }
+    else {
+      sdkEvent.data.isModerator = false;
+    }
+    sdkEvent.data.isMonitor = false;
+    // isRoomHidden
```

Comment:
What is this for?

Reply:
It is a deprecated field that we still return in the webhook payload.   If @joesankey agrees we can leave it out of the payloads generated by the SDK.